### DISCUSSION TITLE
How to make simple functions in symfony?

### DISCUSSION THREAD

**Discussion by Aloxen** [symfony/symfony#51714] [Q&A]
Hey, I know the question must be very odd but I want to make my own functions, that helps me do stuff in my Controller. 

For example if I had written a function that determines whether a number is a power of two or not, how would I include it so I can use it in my Controller? I tried to make a new php file and require_once () it at the top of the Controller. As you probably guessed, that didn't work.

I am looking for the way a professional developer would do it, because I want to learn it the right way from the start on. 

Thank you for your help <3

**Comment by stof** [comment]
The easy way to do that would be to register your file in the `files` section of the composer `autoload` config, but this will load the file each time the autoloader is registered, even if the function is not used (PHP does not have autoloading for functions).

If you want to keep using require_once in your controller to load it only when needed, make sure to use an absolute path (using `__DIR__` as the basis for that). Requiring a relative path won't work well because a relative path is resolved based on the current working directory, which cannot be relied on (when handling web requests, the cwd will be the public folder while when running tests it will generally be the root folder)

**Comment by Aloxen** [comment]
I've read about the autoload feature but I am not sure how I would do that in my case? Would that be called a service and how do I do that exactly?
